drools-persistence-jpa/src/main/java/org/drools/persistence/SingleSessionCommandService.java

```
@@ -400,10 +400,10 @@ private void rollbackTransaction(Exception t1,
             }
             txm.rollback( transactionOwner );
         } catch ( Exception t2 ) {
-            logger.error( "Could not rollback",
-                    t2 );
-            throw new RuntimeException( "Could not commit session or rollback",
-                    t2 );
+            // log rollback exception
+            logger.error( "Could not rollback", t2 );
+            // propagate original exception that caused the rollback
+            throw new RuntimeException( "Could not commit session or rollback", t1 );
```

The message "Could not commit session or rollback" is a bit confusing also for me ( even if I am the last person in the team who should be allowed of speaking about "clear error messages" :) ). Maybe "Could neither commit nor rollback session" is clearer? That said I agree that it's better to propagate the original exception t1 there and just log t2.